### PR TITLE
Updated DocugamiLoader with better support for hierarchical chunks

### PR DESCRIPTION
**Pull request by tjaffri**
There are the following main changes in this PR:

1. Rewrite of the DocugamiLoader to not do any XML parsing of the DGML format internally, and instead use the dgml-utils library we are separately working on. This is a very lightweight dependency.
2. Added MMR search type as an option to multi-vector retriever, similar to other retrievers. MMR is especially useful when using Docugami for RAG since we deal with large sets of documents within which a few might be duplicates and straight similarity based search doesn't give great results in many cases.

We are @docugami on twitter, and I am @tjaffri

### CODE REVIEW

File: libs/langchain/langchain/document_loaders/docugami.py [hunk 1]
```diff
@@ -30,22 +29,51 @@
...
+    To use, you should have the ``dgml-utils`` python package installed.
     """
 
     api: str = DEFAULT_API_ENDPOINT
     """The Docugami API endpoint to use."""
 
     access_token: Optional[str] = os.environ.get("DOCUGAMI_API_KEY")
     """The Docugami API access token to use."""
+
+    max_text_length = 1024 * 4
```
Comment (baskaryan) [inline libs/langchain/langchain/document_loaders/docugami.py:41]: could we just be 4096

File: libs/langchain/langchain/retrievers/multi_vector.py [hunk 2]
```diff
@@ -20,6 +30,8 @@ class MultiVectorRetriever(BaseRetriever):
     id_key: str = "doc_id"
     search_kwargs: dict = Field(default_factory=dict)
     """Keyword arguments to pass to the search function."""
+    search_type: SearchType = SearchType.similarity
```
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/multi_vector.py:33]: ooc will this raise validation error if someone passes in a string? if so would prefer to just type as 
```python
search_type: Literal["similarity", "mmr"] = "similarity"
```

File: poetry.lock
Comment (baskaryan) [inline poetry.lock:1]: don't think this should be updated

File: libs/langchain/langchain/document_loaders/docugami.py [hunk 3]
```diff
@@ -30,22 +29,51 @@
 class DocugamiLoader(BaseLoader, BaseModel):
     """Load from `Docugami`.
 
-    To use, you should have the ``lxml`` python package installed.
+    To use, you should have the ``dgml-utils`` python package installed.
```
[the remark this answers is not in the document]
Comment (tjaffri) [reply libs/langchain/langchain/document_loaders/docugami.py:32]: It is required by dgml-utils so doesn't matter... ref: https://github.com/docugami/dgml-utils/blob/main/python/pyproject.toml#L14

File: poetry.lock
Comment (tjaffri) [reply poetry.lock:1]: I will try a relock... I think your CI checks if the lock file gets out of sync with the .toml so a relock seems to be required, but not sure why I'm getting a merge conflict, will look.

File: libs/langchain/langchain/document_loaders/docugami.py (hunk 1 above)
Comment (tjaffri) [reply libs/langchain/langchain/document_loaders/docugami.py:41]: Sure! sorry probably a style quirk I always try to do it this way to remember that this is 4k etc :)

**Review by baskaryan — APPROVED** [review]

**Pull request merged**